metrics/prometheus.go

```diff
@@ -1581,6 +1582,40 @@ func NewPrometheusCollector(i infoProvider, f ContainerLabelsFunc, includedMetri
 					return values
 				},
 			},
+			{
+				name:        "container_perf_uncore_metric",
+				help:        "Perf uncore event metric.",
+				valueType:   prometheus.CounterValue,
+				extraLabels: []string{"socket", "event", "type"},
+				getValues: func(s *info.ContainerStats) metricValues {
+					values := make(metricValues, 0, len(s.PerfUncoreStats))
+					for _, metric := range s.PerfUncoreStats {
+						values = append(values, metricValue{
+							value:     float64(metric.Value),
+							labels:    []string{machine.GetSocketFromCPU(metric.Cpu), metric.Name, strconv.FormatUint(uint64(metric.Type), 10)},
```

Comment:
It seems like it would be more consistent with other metrics to use the CPU instead of the socket.  Is there a particular reason why we are using socket here?

Reply:
Uncore PMUs (Performance Monitoring Units) are inside CPU but outside of cores. This means that they count events per the whole CPU.
To count events we need to call ``perf_event_open`` with CPU argument, so we use CPUs from ``cpumask`` ( for ex. ``/sys/devices/uncore_imc_0/cpumask`` ). These CPUs correspond to Sockets, and there is only one CPU per Socket.